qdlt/tests/test_qdltargument.cpp

```diff
@@ -0,0 +1,71 @@
+#include <gtest/gtest.h>
+
+#include <qdltargument.h>
+
+TEST(QDltArgument, constructor) {
+    QDltArgument arg;
+
+    ASSERT_EQ(arg.getTypeInfo(), QDltArgument::DltTypeInfoUnknown);
+    ASSERT_EQ(arg.getTypeInfoString(), QString{});
+    ASSERT_EQ(arg.getDataSize(), 0);
+    ASSERT_EQ(arg.toString(), QString("?")); // a bit weird for data size 0
+    ASSERT_EQ(arg.getValue(), QVariant());
+}
+
+TEST(QDltArgument, string_ascii) {
+    QDltArgument arg;
+    arg.setTypeInfo(QDltArgument::DltTypeInfoStrg);
+    const unsigned char asc_data[] = {
+        0xc3, 0xbc, 'b', 'e' // Ã¼be (and not übe)
+    };
+    arg.setData(QByteArray::fromRawData((const char*)asc_data, sizeof(asc_data)));
+    ASSERT_EQ(arg.toString(), QString::fromUtf8("Ã¼be"));
+
+    // parse from raw payload
+    const unsigned char asc_data2[] = {0x00, 0x02, 0x00, 0x00, // type info
+                                       0x04, 0x00,             // str len
+                                       0xc3, 0xbc, 'b',  'e'};
+    QByteArray payload = QByteArray::fromRawData((const char*)asc_data2, sizeof(asc_data2));
+    unsigned int offset = 0;
+    arg.setArgument(payload, offset, QDlt::DltEndiannessLittleEndian);
+    ASSERT_EQ(arg.getTypeInfo(), QDltArgument::DltTypeInfoStrg);
+    ASSERT_EQ(arg.getDataSize(), 4);
+    ASSERT_EQ(arg.toString().length(), 4);
+    ASSERT_EQ(arg.toString(), QString::fromUtf8("Ã¼be"));
+    ASSERT_EQ(arg.toString(true), QString{"c3 bc 62 65"});
+    ASSERT_EQ(arg.getValue(), QVariant(QString::fromUtf8("Ã¼be")));
+}
+
+TEST(QDltArgument, string_utf8) {
+    QDltArgument arg;
+    arg.setTypeInfo(QDltArgument::DltTypeInfoUtf8);
+    const unsigned char asc_data[] = {
+        0xc3, 0xbc, 'b', 'e' // übe
+    };
+    arg.setData(QByteArray::fromRawData((const char*)asc_data, sizeof(asc_data)));
+    ASSERT_EQ(arg.toString(), QString::fromUtf8((const char*)asc_data, sizeof(asc_data)));
+
+    // parse from raw payload
+    const unsigned char asc_data2[] = {0x00, 0x82, 0x00, 0x00, // type info
```

Comment:
maybe create QByteArray directly? this way you will save on casting from raw C-arrays

Reply:
QByteArray has the same (signed) char instead of unsigned char constructor issue, doesn't it?

Reply:
I thought about something like this: 
```
QByteArray payload = QByteArray::fromHex("000200000400c3bc6265");
```

Reply:
@mbehr1 would you mind making it nicer as I suggested?

Reply:
@vifactor I thought about it but I find it more difficult to read as an indenting and commenting of byte slices would not be so easy.
I could change to a 
"...fromHex("000" // foo
 + "1234" // bar
+ "...." // xyz
); 

But that's not really nice either?

Reply:
ok, fair enough